plugins/packages/rethinkdb/lib/index.ts

```diff
@@ -0,0 +1,268 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-plugins/common';
+import { SourceOptions, QueryOptions } from './types';
+const JSON5 = require('json5');
+const r = require('rethinkdb');
+
+export default class Rethinkdb implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions): Promise<QueryResult> {
+    const operation = queryOptions.operation;
+    const connection = await this.getConnection(sourceOptions);
+    let result: any = {};
+    const { tablename, name, key, body } = queryOptions;
+    const { database } = sourceOptions;
+    let response = null;
+
+    try {
+      switch (operation) {
+        case 'db_create': {
+          result = await this.createDatabase(name, connection);
+          break;
+        }
+        case 'db_drop': {
+          result = await this.dropDatabase(name, connection, database);
+          break;
+        }
+        case 'create_table': {
+          result = await this.createTable(name, tablename, connection, database);
+          break;
+        }
+        case 'drop_table': {
+          result = await this.dropTable(name, tablename, connection, database);
+          break;
+        }
+        case 'db_list': {
+          result = await this.listAllDatabase(connection, database);
+          break;
+        }
+        case 'list_table': {
+          result = await this.listAllTables(name, connection, database);
+          break;
+        }
+        case 'list_documents': {
+          response = await this.listAllDocuments(name, tablename, connection, database);
+          result = await response.toArray(function (err, cursor) {
+            if (err) throw err;
+            return cursor;
+          });
+          break;
+        }
+        case 'create_docs': {
+          result = await this.insertDocument(name, tablename, body, connection, database);
+          break;
+        }
+        case 'retreive_docs': {
+          result = await this.getDocumentByID(name, tablename, key, connection, database);
+          break;
+        }
+
+        case 'update_docs': {
+          result = await this.listAllTables(name, connection, database);
+          break;
```

Comment:
This case is conflicting with `list_table` case

Reply:
updated @gondar00